packages/augur-state/src/api/Users.ts

```diff
@@ -1,30 +1,222 @@
-// TODO: Define Users class
-
-export interface GetUserTradingPositionsParams {
-  universe: string,
-  creator?: string,
-  category?: string,
-  search?: string,
-  reportingState?: string,
-  disputeWindow?: string,
-  designatedReporter?: string,
-  maxFee?: number,
-  hasOrders?: boolean,
+import {SortLimit} from './types';
+import { DB } from "../db/DB";
+import * as _ from "lodash";
+import { Dictionary, NumericDictionary } from "lodash";
+import { numTicksToTickSize, convertOnChainAmountToDisplayAmount, convertOnChainPriceToDisplayPrice } from "@augurproject/api";
+import { BigNumber } from "bignumber.js";
+import { TrackedUsers } from '../db/TrackedUsers';
+import { ProfitLossChangedLog, OrderFilledLog } from '../logs/types';
+
+export interface UserTradingPositionsParams {
+  universe?: string,
+  account?: string,
+  marketId?: string,
+  outcome?: number,
+};
+
+export interface GetUserTradingPositionsParams extends UserTradingPositionsParams, SortLimit {
+}
+
+export interface GetProfitLossParams {
+  universe?: string,
+  account?: string,
+  startTime?: number,
+  endTime?: number,
+  periodInterval?: number,
+  marketId?: string,
+  outcome?: number,
+};
+
+
+export interface GetProfitLossSummaryParams {
+  universe?: string,
+  account?: string,
+  marketId?: string,
+  endTime?: number,
+};
+
+export interface MarketTradingPosition {
+  timestamp: number;
+  frozenFunds: string;
+  marketId: string;
+  realized: string; // realized profit in tokens (eg. ETH) user already got from this market outcome. "realized" means the user bought/sold shares in such a way that the profit is already in the user's wallet
+  unrealized: string; // unrealized profit in tokens (eg. ETH) user could get from this market outcome. "unrealized" means the profit isn't in the user's wallet yet; the user could close the position to "realize" the profit, but instead is holding onto the shares. Computed using last trade price.
+  total: string; // total profit in tokens (eg. ETH). Always equal to realized + unrealized
+  unrealizedCost: string; // denominated in tokens. Cost of shares in netPosition
+  realizedCost: string; // denominated in tokens. Cumulative cost of shares included in realized profit
+  totalCost: string; // denominated in tokens. Always equal to unrealizedCost + realizedCost
+  realizedPercent: string; // realized profit percent (ie. profit/cost)
+  unrealizedPercent: string; // unrealized profit percent (ie. profit/cost)
+  totalPercent: string; // total profit percent (ie. profit/cost)
+  currentValue: string; // current value of netPosition, always equal to unrealized minus frozenFunds
+}
+
+export interface TradingPosition {
```

Comment:
Might also want to consider having MarketTradingPostion & TradingPosition extend another interface to minimize redundancy.

Reply:
Yea planning to do this once I get to those getters